src/habitat.cr

```diff
@@ -37,6 +37,64 @@ class Habitat
     end
   end
 
+  # Embed settings in a Class or Module.
+  #
+  # A class or module can call `Habitat.create` with a block of `setting` calls
+  # that will declare the types (and optionally default values) of our settings.
+  #
+  # ```
+  # class MyServer
+  #   Habitat.create do
+  #     setting port : Int32
+  #     setting debug_errors : Bool = true
+  #   end
+  # end
+  # ```
+  #
+  # `create` adds class and instance `settings` methods to the embedding
+  # class/module, which we'll use to set and get the values of our settings.
+  #
+  # ```
+  # MyServer.settings.port = 80
```

Comment:
Though this is possible I think it is best to stick to using `MyServer.configure`. I'd have to play around with things a bit more before seeing if we should also recommend calling `settings =` directly

Reply:
Gotcha. I guess you'd also prefer to keep the instance-level `settings` setter undocumented for now as well?

Reply:
Yes I think leaving those undocumented would be perfect 👍